### ISSUE TITLE
vue/define-emits-declaration Does not work in setup lang='tsx'

### ISSUE THREAD

**Issue by laterdayi** [vuejs/eslint-plugin-vue#2098]
**Checklist**

- [√ ] I have tried restarting my IDE and the issue persists.
- [√ ] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: lastet
- **eslint-plugin-vue version:lastet
- **Node version:lastet
- **Operating System:windows

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true,
    es6: true
  },
  parser: "vue-eslint-parser",
  plugins: ["standard-recommended"],
  extends: [
    "plugin:vue/vue3-essential",
    "plugin:vue/vue3-strongly-recommended",
    "plugin:vue/vue3-recommended",
    "eslint:recommended",
    "@vue/eslint-config-typescript/recommended",
    "@vue/eslint-config-prettier",
    "./.eslintrc-auto-import.json"
  ],
  parserOptions: {
    parser: "@typescript-eslint/parser",
    ecmaVersion: "latest",
    ecmaFeatures: {
      jsx: true
    }
  },
  noInlineConfig: true,
  reportUnusedDisableDirectives: true,
  rules: {
    "standard-recommended/no-index": "error",
    "prettier/prettier": ["warn", { printWidth: 120 }],
    "no-console": process.env.NODE_ENV === "production" ? "warn" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "warn" : "off",
    eqeqeq: "error",
    "no-alert": "error",
    quotes: "off",
    "no-const-assign": "error",
    "no-eval": "error",
    "no-implied-eval": "error",
    semi: "off",
    "no-multi-spaces": "error",
    "no-new-wrappers": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-useless-return": "error",
    "no-void": "error",
    "require-await": "error",
    "wrap-iife": ["error", "inside"],
    yoda: "error",
    "no-undef": "off",
    "no-use-before-define": "off",
    "global-require": "error",
    "no-new-require": "error",
    camelcase: "error",
    "comma-dangle": "error",
    "line-comment-position": "error",
    "max-depth": "error",
    "max-nested-callbacks": "error",
    "multiline-comment-style": "error",
    "no-inline-comments": "error",
    "no-lonely-if": "error",
    "no-trailing-spaces": "error",
    "no-unneeded-ternary": "error",
    "no-whitespace-before-property": "error",
    "operator-assignment": "error",
    "padded-blocks": ["error", "never"],
    "space-before-function-paren": "off",
    "spaced-comment": "error",
    "switch-colon-spacing": "error",
    "no-duplicate-imports": "error",
    "no-useless-rename": "error",
    "no-lone-blocks": "error",
    "no-return-await": "error",
    "array-bracket-spacing": "error",
    "arrow-parens": ["error", "as-needed"],
    "arrow-spacing": "error",
    "brace-style": "error",
    "comma-spacing": "error",
    "key-spacing": "error",
    "keyword-spacing": "error",
    "lines-between-class-members": "error",
    "no-multiple-empty-lines": "error",
    "nonblock-statement-body-position": "error",
    "object-curly-spacing": ["error", "always"],
    "rest-spread-spacing": "error",
    "space-before-blocks": "error",
    "space-in-parens": "error",
    "@typescript-eslint/array-type": "error",
    "@typescript-eslint/no-duplicate-enum-values": "error",
    "@typescript-eslint/no-empty-interface": "error",
    "@typescript-eslint/no-inferrable-types": "error",
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/no-explicit-any": "error",
    "@typescript-eslint/no-namespace": "off",
    "@typescript-eslint/prefer-optional-chain": "error",
    "@typescript-eslint/naming-convention": [
      "error",
      { selector: ["interface", "typeAlias", "enum"], format: ["PascalCase"] },
      { selector: ["function", "parameter"], format: ["camelCase"] },
      {
        selector: "memberLike",
        modifiers: ["private"],
        format: ["camelCase"],
        leadingUnderscore: "require"
      }
    ],
    "vue/space-in-parens": "error",
    "vue/object-curly-spacing": ["error", "always"],
    "vue/keyword-spacing": "error",
    "vue/key-spacing": "error",
    "vue/camelcase": "error",
    "vue/brace-style": "error",
    "vue/arrow-spacing": "error",
    "vue/array-bracket-spacing": "error",
    "vue/eqeqeq": "error",
    "vue/v-for-delimiter-style": "error",
    "vue/static-class-names-order": "error",
    "vue/require-prop-comment": ["error", { type: "line" }],
    "vue/require-direct-export": "error",
    "vue/prefer-true-attribute-shorthand": "error",
    "vue/prefer-separate-static-class": "error",
    "vue/padding-lines-in-component-definition": "error",
    "vue/padding-line-between-blocks": "error",
    "vue/no-v-text": "error",
    "vue/no-useless-v-bind": "error",
    "vue/no-useless-mustaches": "error",
    "vue/no-static-inline-styles": "error",
    "vue/html-comment-indent": "error",
    "vue/html-comment-content-spacing": "error",
    "vue/html-comment-content-newline": "error",
    "vue/define-emits-declaration": "error",
    "vue/custom-event-name-casing": ["error", "kebab-case"],
    "vue/component-options-name-casing": ["error", "PascalCase"],
    "vue/component-api-style": "error",
    "vue/block-tag-newline": "error",
    "vue/this-in-template": "error",
    "vue/order-in-components": "error",
    "vue/no-lone-template": "error",
    "vue/component-tags-order": [
      "error",
      {
        order: ["script:not([setup])", "script[setup]", "template", "style"]
      }
    ],
    "vue/v-slot-style": "error",
    "vue/v-on-style": "error",
    "vue/v-on-event-hyphenation": "error",
    "vue/v-bind-style": "error",
    "vue/require-prop-types": "error",
    "vue/require-explicit-emits": "error",
    "vue/require-default-prop": "error",
    "vue/prop-name-casing": "error",
    "vue/one-component-per-file": "error",
    "vue/no-template-shadow": "error",
    "vue/no-spaces-around-equal-signs-in-attribute": "error",
    "vue/no-multi-spaces": "error",
    "vue/mustache-interpolation-spacing": "error",
    "vue/html-indent": "error",
    "vue/html-self-closing": [
      "error",
      {
        html: {
          void: "always",
          normal: "always",
          component: "always"
        },
        svg: "always",
        math: "always"
      }
    ],
    "vue/attribute-hyphenation": "error",
    "vue/no-setup-props-destructure": "off",
    "vue/multi-word-component-names": "off",
    "vue/no-v-html": "off"
  },
  overrides: [
    {
      files: [
        "src/directive/index.ts",
        "src/plugins/core/index.ts",
        "src/plugins/vite/plugins/index.ts",
        "src/service/index.ts",
        "src/router/index.ts"
      ],
      rules: {
        "standard-recommended/no-index": "off"
      }
    },
    {
      files: ["src/**/*"],
      rules: {
        "standard-recommended/file-name-convention": "error",
        "standard-recommended/folder-name-convention": "error"
      }
    },
    {
      files: ["src/App.vue", "src/AppProvider.vue"],
      rules: {
        "standard-recommended/file-name-convention": "off"
      }
    },
    {
      files: [
        "src/service/type.ts",
        "src/components/permission-button/type.ts",
        "src/components/base-chart/**/*",
        "src/components/base-curd/**/*",
        "src/components/base-ui/**/*",
        "src/types/global.d.ts",
        "src/utils/component.ts",
        "src/utils/extends.ts",
        "src/utils/localstorage.ts"
      ],
      rules: {
        "@typescript-eslint/no-explicit-any": "off"
      }
    },
    {
      files: ["env.d.ts"],
      rules: {
        "multiline-comment-style": "off",
        "spaced-comment": "off"
      }
    }
  ]
};

```

**What did you do?**	

```vue

I've set vue/define-emits-declaration, but it doesn't work with setup and lang='tsx'

<script setup lang="tsx">


const emit = defineEmits(['reset-table']);

</script>

```

![image](https://user-images.githubusercontent.com/122137647/222179670-7afb32eb-94bd-460e-b843-a1c7dbad9841.png)

**What did you expect to happen?**

It can indicate errors

**What actually happened?**

**Repository to reproduce this issue**
